### DISCUSSION TITLE
Support SQLite in Cartesi node

### DISCUSSION THREAD

**Discussion by gligneul** [cartesi/rollups-node#156] [General]
## 📚 Context

To run the node, we need to start a PostgreSQL database. This is great for deploying the node in the cloud, but cumbersome when running it locally. It increases the startup time and the processing requirement to run the node. 

## ✔️ Solution

Add support for SQLite as the rollups database. The indexer should store the data in SQLite, and the GraphQL server should read from it.

This should be a CLI/ENV option; the Postgres option should also exist.

## 📈 Subtasks

- [ ] Design solution and write the next subtasks.

**Comment by endersonmaia** [comment]
As a suggestion, we could parse a `DATABASE_URL` and detect it by protocol.

`<protocol>://[<user>][:password>@<host>[:<port>]/[<database>]`

and support:

- `postgresql://...`
- `sqlite:///path/to/file.sqlite`

**Comment by gligneul** [comment]
I'm migrating this to a discussion. Once we have the Go node in place, I will create the corresponding issue.

**Comment by endersonmaia** [comment]
I strongly suggest supporting SQLite in an attempt to have a rollups-node in a single binary without any external dependency.

This could make the adoption and deployment way easy than it is today.

I know we have postgraphile and subsquid ideas to reduce the amount of code needed to make the conversion from rust to new golang codebase, and I'm assuming they are transitory.

Take as an example this project: https://pocketbase.io/docs/

Download a single binary for your platform, and run `pocketbase server`.

Maybe this could be the future goal of the golang rollups-node.

**Reply by endersonmaia** [reply]
Add to it the possibility of running the rollups-node on the browser together with the cartesi-machine using WASM, sqlite would be essential.

**Reply by endersonmaia** [reply]
And just after my last comment, I see this https://github.com/electric-sql/pglite

It's experimental, but worth the bookmark.
